log/test.ts

```diff
@@ -1,28 +1,137 @@
-import { remove, open, readAll } from "deno";
 import { assertEqual, test } from "../testing/mod.ts";
 import * as log from "./mod.ts";
-import { FileHandler } from "./handlers.ts";
+import { BaseHandler } from "./handlers.ts";
+import { LogRecord } from "./logger.ts";
+import { LogLevel } from "./levels.ts";
 
-// TODO: establish something more sophisticated
-let testOutput = "";
+class TestHandler extends BaseHandler {
+  public messages: string[] = [];
+  public records: LogRecord[] = [];
```

Comment:
Why do we need it?

Reply:
we assert that the correct records are added in a test further down. otherwise we'd only be asserting on the log messages

Reply:
Handler's main responsibility is to output message, so checking only outputted messages should be enough. There's even a TODO that suggested freezing log records after they're created so handlers can't mess them up.

Reply:
There should still be a test for these records, though, as custom handlers should be called correctly with the records.

I've moved the test to the logger tests file as it is rather a specific assertion there: "if we call the logger's log function, a record is passed to our handler".